cmd_line_tools/commands.py

```diff
@@ -1,11 +1,195 @@
+import random
+from requests.exceptions import HTTPError
+
 from .mixins import (
     SimpleCommandLineParserMixin, ArgumentsRequestMixin, StdoutOutputMixin,
     InputRequestMixin, SimpleAuthenticationMixin,
-    LoginMixin)
+    LoginMixin, JSONDataRequestMixin, PagedJSONDataMixin, CSVOutputMixin,
+    PrintURLPageTitleToStdoutMixin, GoogleFeelingLuckyMixin, OpenURLInBrowser,
+    RequestValidatorMixin, ValidationError)
 
 __all__ = [
     'ArgumentCalculatorCommand', 'InputCalculatorCommand',
-    'PriviledgedArgumentsExampleCommand']
+    'PriviledgedArgumentsExampleCommand', 'InputBasedPokemon',
+    'CommandLinePokemon', 'RandomChooserPokemon', 'PokemonBerryPrinter',
+    'PokemonBerryCSVWriter', 'StdoutArgsFeelingLucky', 'ValidatedInputCommand']
+
+
+class ValidatedInputCommand(InputRequestMixin,
+                            RequestValidatorMixin,
+                            StdoutOutputMixin):
+
+    def __init__(self, validator=None):
+        self.validator = validator
+
+    def main(self):
+        self.request_input_data('data')
+        try:
+            result = self.validate(validator=self.validator)
+        except ValidationError as e:
+            result = str(e)
+        self.write(result)
+
+
+class BaseFeelingLucky(object):
+    """
+    Takes a search term and returns the Google I'm Feeling Lucky result
+    """
+
+    def feeling_lucky(self):
+        search_term = self.request_input_data('search_term')
+        url = self.request_data(search_term)
+        self.open(url)
+
+
+class StdoutArgsFeelingLucky(BaseFeelingLucky,
+                             GoogleFeelingLuckyMixin,
+                             ArgumentsRequestMixin,
+                             SimpleCommandLineParserMixin,
+                             PrintURLPageTitleToStdoutMixin):
+    """ Prints resulting page's title and url to stdout """
+
+    def main(self):
+        self.parse_arguments()
+        self.feeling_lucky()
+
+
+class OpenInBrowserArgsFeelingLucky(BaseFeelingLucky,
+                                    GoogleFeelingLuckyMixin,
+                                    ArgumentsRequestMixin,
+                                    SimpleCommandLineParserMixin,
+                                    OpenURLInBrowser):
+    """ Opens resulting url in web browser """
+
+    def main(self):
+        self.parse_arguments()
+        self.feeling_lucky()
+
+
+class BaseAllPokemonBerriesCommand(object):
+    """
+    Grabs entries for all berries from PokeAPI
+    and then outputs them
+    """
+    REQUEST_URL = 'http://pokeapi.co/api/v2/berry/'
+
+    def get_pokemon(self):
+        return self.request_data()
+
+
+class PokemonBerryPrinter(PagedJSONDataMixin,
+                          StdoutOutputMixin,
+                          BaseAllPokemonBerriesCommand):
+    """ Prints to stdout """
+
+    def main(self):
+        results = self.get_pokemon()
+        for berry in results:
```

Comment:
My pokeknowledge is rather limited, though I like the idea of your project. Am just curious, though... "berry"?